### PR TITLE
[FLINK-2827] Closing FileInputStream through try-with-resources to avoid unused open stream.

### PR DESCRIPTION
**Pull request by ssaumitra**

### CODE REVIEW

File: flink-staging/flink-streaming/flink-streaming-connectors/flink-connector-twitter/src/main/java/org/apache/flink/streaming/connectors/twitter/TwitterSource.java [hunk 1]
```diff
@@ -133,13 +134,21 @@ protected OAuth1 authenticate() {
...
 		try {
-			InputStream input = new FileInputStream(authPath);
+			input = new FileInputStream(authPath);
 			properties.load(input);
-			input.close();
 		} catch (Exception e) {
 			throw new RuntimeException("Cannot open .properties file: " + authPath, e);
 		}
+		finally {
+			try {
```
Comment (ssaumitra) [inline flink-staging/flink-streaming/flink-streaming-connectors/flink-connector-twitter/src/main/java/org/apache/flink/streaming/connectors/twitter/TwitterSource.java]: This try..catch does not look beautiful. But `input.close()` mandates it. Also `close()` will not be called if `FileInputStream` is not open.

**Comment by StephanEwen** [comment]
How about using a "try with resources" statement?

**Comment by mxm** [comment]
+1 We're on Java 7, which allows to implicitly close resources in a try statement:

**Comment by ssaumitra** [comment]
I just observed that Flink has already dropped Java 6 support. try with resources is the best possible approach then. I will amend my commit.

**Comment by ssaumitra** [comment]
@StephanEwen I have made the changes and rebased on master. Kindly review.

**Comment by mxm** [comment]
Looks good!

**Comment by fhueske** [comment]
I'm going to merge this PR.

**Pull request closed without merging**